spark3/src/test/java/org/apache/iceberg/actions/TestCreateActions.java

```diff
@@ -157,6 +168,105 @@ public void testMigratePartitioned() throws Exception {
     assertMigratedFileCount(Actions.migrate(source), source, dest);
   }
 
+  @Test
+  public void testPartitionedTableWithUnRecoveredPartitions() throws Exception {
```

Comment:
What is this testing? I'm not familiar with what unrecovered partitions are.

Reply:
Example usecase:
There is  a location where a external system is writing files every day.
An external partitioned Hive table is created on top of the location, partitioned on day.
Once the external table is created, we need to run "ALTER TABLE RECOVER PARTITIONS" or "MSCK REPAIR TABLE" to register the partitions in metastore. .

When we dont recover(register against metastore) a partitions(intentionally or unintentionally), the data corresponding to unrecovered partition is not returned as part of the table scan.
This test checks if both Spark and Iceberg returns same data, when there are unrecovered partitions

Reply:
Sounds like a good thing to test.